### PR TITLE
Fix symbolic pass for printing `ListItem`

### LINKED ISSUE #2613: Negative indices in list fails
```console
% cat examples/expr2.py 
from lpython import i32

A: list[i32] = [1, 2, 4, 5]
print(A[0], A[1], A[2], "...", A[-3], A[-2], A[-1])
% python examples/expr2.py 
1 2 4 ... 2 4 5
% lpython examples/expr2.py
...
```

### PR DESCRIPTION
**Pull request by kmr-srbh**
fixes #2613 

```python
from lpython import i32

A: list[i32] = [1, 2, 4, 5]
print(A[0], A[1], A[2], "...", A[-3], A[-2], A[-1])
```
```console
(base) saurabh-kumar@Awadh:~/Projects/System/lpython$ ./src/bin/lpython ./examples/example.py
1 2 4 ... 2 4 5
```

### CODE REVIEW

**Review by Shaikh-Ubaid — COMMENTED** [review]
Add an integration test for this with print and assert statements.

File: src/libasr/pass/replace_symbolic.cpp [hunk 1]
```diff
@@ -842,6 +842,11 @@ class ReplaceSymbolicVisitor : public PassUtils::PassVisitor<ReplaceSymbolicVisi
                         x.base.base.loc, list_item->m_a, list_item->m_pos,
                         ASRUtils::TYPE(ASR::make_CPtr_t(al, x.base.base.loc)), nullptr));
                     print_tmp.push_back(basic_str(x.base.base.loc, value));
+                } else {
+                    ASR::expr_t *value = ASRUtils::EXPR(ASR::make_ListItem_t(al,
+                        x.base.base.loc, list_item->m_a, list_item->m_pos,
+                        ASRUtils::expr_type(val), nullptr));
+                    print_tmp.push_back(value);
```
Comment (Shaikh-Ubaid) [inline src/libasr/pass/replace_symbolic.cpp:849]: Let's simply push_back `val`.
```suggestion
                    print_tmp.push_back(val);
```

File: integration_tests/test_list_item_mixed_print.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    for i in range(len(i_list)):
+        print(i_list[i], end=" # ")
+
+    n_list: list[list[i32]] = [[1, 2], [3, 4], [5, 6]]
+    for i in range(len(n_list)):
+        print("List ", i, ":", n_list[i])
+
+test_list_item_mixed_print()
+
+
```
Comment (Shaikh-Ubaid) [inline integration_tests/test_list_item_mixed_print.py:45]:
```suggestion
```

File: integration_tests/test_list_item_mixed_print.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    i_list2: list[i32] = [1, 2, 3]
+    print(i_list2[0], i_list2[1], i_list2[2], sep=" is smaller than ")
+
+    i: i32
+    for i in range(len(i_list)):
+        print(i_list[i], end=" # ")
+
+    n_list: list[list[i32]] = [[1, 2], [3, 4], [5, 6]]
+    for i in range(len(n_list)):
+        print("List ", i, ":", n_list[i])
```
Comment (Shaikh-Ubaid) [inline integration_tests/test_list_item_mixed_print.py:41]: Please add assert statements after each print statement in this test.

**Review by Shaikh-Ubaid — COMMENTED** [review]
Thanks for this! I shared few comments. It looks good overall.

File: integration_tests/test_list_item_mixed_print.py (hunk 3 above)
Comment (kmr-srbh) [reply integration_tests/test_list_item_mixed_print.py:41]: @Shaikh-Ubaid I am a bit confused with this. Could you please guide me on what should be asserted? We are only printing. I did not find assert statements in any other test for printing.

Comment (Shaikh-Ubaid) [reply integration_tests/test_list_item_mixed_print.py:41]: Ok, it is fine. Let's also test this in reference test with `run = true`.

Comment (kmr-srbh) [reply integration_tests/test_list_item_mixed_print.py:42]: Done.

**Review by Shaikh-Ubaid — APPROVED** [review]
It looks good! Thanks! Great work!

**Pull request merged**